compiler/src/iree/compiler/Codegen/Common/VectorReductionToGPU.cpp

```diff
@@ -40,28 +42,77 @@ static Value allocateGlobalSharedMemory(Location loc, OpBuilder &builder,
   return builder.create<memref::AllocOp>(loc, memrefType);
 }
 
+/// Packs vector of lower precision into a single 32-bit width element.
+/// (i.e <2xf16> -> f32 and <4xi8> -> i32)
+static Value packVectorToSupportedWidth(Location loc, OpBuilder &builder,
+                                        Value input) {
+  auto vecType = input.getType().cast<VectorType>();
+  assert(vecType && "This packing fn only works if input is a vector.");
```

Comment:
nit: `cast` already asserts that the cast is correct and never return null

Reply:
done!